### PR TITLE
insync: 3.8.5.50499 -> 3.8.6.50504

### PR DESCRIPTION
**Pull request by hellwolf**
###### Description of changes

Upgrading to [insync 3.8.6.50504](https://forums.insynchq.com/t/new-insync-version-3-8-6/19215).

With fixes:

- Top-level runner is now simply insync, in sync with vendor's desktopfile. <--- addressing a comment in https://github.com/NixOS/nixpkgs/issues/139728
- The /share folder including desktop files are now part of the top package.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by hellwolf** [comment]
@ofborg build insync

**Comment by hellwolf** [comment]
Hmm, why is https://github.com/NixOS/nixpkgs/pull/233044/checks?check_run_id=13630890667 not evaluated?

**Comment by benwbooth** [comment]
Thanks for this! Can you please also add the contents of /share/icons?

**Comment by benwbooth** [comment]
/share/man would be good to in order to have man pages for the cli commands.

**Comment by hellwolf** [comment]
> Thanks for this! Can you please also add the contents of /share/icons?
> /share/man would be good to in order to have man pages for the cli commands.

Actually I might just expose all `/share` from insync vendor package:

```
$ tree -L 2  ./result/share
./result/share
├── applications
│   ├── insync.desktop
│   └── insync-helper.desktop
├── doc
│   └── insync
├── icons
│   └── hicolor
├── man
│   └── man1
└── mime
    └── packages
```

Btw, what are you looking from having /share/icons exposed in the top-level package? I am still getting nonetheless:

```
$ ./result/bin/insync start --no-daemon
QSystemTrayIcon::setVisible: No Icon set
```

**Comment by benwbooth** [comment]
The icons need to be exposed in order for desktop environments to find them. That way it will show the insync icon when you put the launcher in the dock or taskbar.

**Comment by hellwolf** [comment]
> The icons need to be exposed in order for desktop environments to find them. That way it will show the insync icon when you put the launcher in the dock or taskbar.

I meant the SystemTrayIcon, they were not working for me.

I am yet to upgrade to 23.05, I hope to do that shortly after the release. So I cannot actually fully test this package apart from running the `./result/bin/insync` for now. If you could test and report if the known bugs are fixed, that'd be great! If not it's not a problem neither, I will give it another short another time!

File: pkgs/applications/networking/insync/default.nix [hunk 1]
```diff
@@ -81,30 +82,43 @@ let
 
     # NB! This did the trick, otherwise it segfaults! However I don't understand why!
     dontStrip = true;
+  };
 
+  insync-fhsenv = buildFHSEnv { # ref: pkgs/build-support/build-fhsenv-bubblewrap/default.nix
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/insync/default.nix:87]:
```suggestion
  insync-fhsenv = buildFHSEnv {
```
that comment will eventually be outdated

File: pkgs/applications/networking/insync/default.nix [hunk 2]
```diff
@@ -81,30 +82,43 @@ let
...
     # QT_STYLE_OVERRIDE was used to suppress a QT warning, it should have no actual effect for this binary.
     export QT_STYLE_OVERRIDE=Fusion
     exec "${insync-pkg.outPath}/lib/insync/insync" "$@"
+    '';
+
+    # "insync start" command starts a daemon.
+    dieWithParent = false;
+  };
+
+in stdenv.mkDerivation {
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/insync/default.nix:113]:
```suggestion
in stdenvNoCC.mkDerivation {
```

**Comment by hellwolf** [comment]
Thanks @SuperSandro2000, incorporated your changes, especially now using stdenvNoCC in both occasions.

I haven't found the original document of it, but I assume it means what it says on the label...

File: pkgs/applications/networking/insync/default.nix [hunk 3]
```diff
@@ -81,30 +82,43 @@ let
...
+    # "insync start" command starts a daemon.
+    dieWithParent = false;
+  };
+
+in stdenvNoCC.mkDerivation {
+  inherit pname version meta;
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/insync/default.nix:118]:
```suggestion

```
isn't this a noop?

**Comment by aaylnx** [comment]
@hellwolf Thank you very much for your work on this!

I am running nix on Fedora Silverblue and I was getting this error when running `insync start`

Doing a little bit of googling I found [NixGL](https://github.com/guibou/nixGL) which may be necessary for people running nix on something other than NixOS.

After doing
` nix-channel --add https://github.com/guibou/nixGL/archive/main.tar.gz nixgl && nix-channel --update
  nix-env -iA nixgl.auto.nixGLDefault
 `

I was able to get insync running with

`nixGL insync start`

However, while an entry shows up in the KDE systemtray for Insync, there's no icon there. I can open up the insync app window if I right click on the entry in the systemtray, but it would be nice if I could get an icon to show up.

If anyone knows how to address that, please chime in.

Thanks again!

**Comment by hellwolf** [comment]
> @hellwolf Thank you very much for your work on this!
> 
> I am running nix on Fedora Silverblue and I was getting this error when running `insync start`
> 
> Doing a little bit of googling I found [NixGL](https://github.com/guibou/nixGL) which may be necessary for people running nix on something other than NixOS.
> 
> After doing `nix-channel --add https://github.com/guibou/nixGL/archive/main.tar.gz nixgl && nix-channel --update ; nix-env -iA nixgl.auto.nixGLDefault`
> 
> I was able to get insync running with
> 
> `nixGL insync start`
> 
> However, while an entry shows up in the KDE systemtray for Insync, there's no icon there. I can open up the insync app window if I right click on the entry in the systemtray, but it would be nice if I could get an icon to show up.
> 
> If anyone knows how to address that, please chime in.
> 
> Thanks again!

Thanks for the information. That is interesting, I haven't heard that before. Could it be Fedora Silverblue specific? 

Also re icon not shown issue, it had been reported here: https://github.com/NixOS/nixpkgs/issues/244813

**Comment by aaylnx** [comment]
@hellwolf, Thanks very much for the reply. Yes, I think the bug that you link to is my issue and not something Silverblue specific. If I run `insync status`, I also get the same reply reported there, "Insync doesn't seem to be running. Start it first." Nevertheless, it is running and I can access it via the kde system tray, however there's simply a blank space for the icon. As you indicate, this is most likely an upstream bug. Thanks!

**Pull request merged**